Neos.ContentRepository.Core/Classes/Feature/ContentStreamCommandHandler.php

```diff
@@ -130,14 +129,13 @@ private function handleReopenContentStream(
      */
     private function handleForkContentStream(
         ForkContentStream $command,
-        ContentRepository $contentRepository
+        CommandHandlingDependencies $commandHandlingDependencies
     ): EventsToPublish {
-        $this->requireContentStreamToExist($command->sourceContentStreamId, $contentRepository);
-        $this->requireContentStreamToNotBeClosed($command->sourceContentStreamId, $contentRepository);
-        $this->requireContentStreamToNotExistYet($command->newContentStreamId, $contentRepository);
+        $this->requireContentStreamToExist($command->sourceContentStreamId, $commandHandlingDependencies);
+        $this->requireContentStreamToNotBeClosed($command->sourceContentStreamId, $commandHandlingDependencies);
 
-        $sourceContentStreamVersion = $contentRepository->getContentStreamFinder()
-            ->findVersionForContentStream($command->sourceContentStreamId);
+        // TOOD: THis is not great
```

Comment:
```suggestion
        // TODO: This is not great
```

and: why is it not great?